### DISCUSSION TITLE
need to add port 7800

### DISCUSSION THREAD

**Discussion by ramu11** [arkmq-org/arkmq-org-broker-operator#491] [Ideas] [answered]
I'm able to get the AMQ cluster to form by adding the the traffic.sidecar.istio.io/excludeInboundPorts: "7800" annotation on the pod template in the statefulset created by the operator.

I feel like the root of the issue is that port 7800 is not listed in the headless service created by the operator (only ports 8161 and 61616 are listed), even though it's used by the pods to communicate directly with each other via jgroup. Would it make sense to create an RFE to get that port added?

oc get svc/amq-hdls-svc
NAME           TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)              AGE
amq-hdls-svc   ClusterIP   None         <none>        8161/TCP,61616/TCP   43d
Quoting the istio doc: https://istio.io/latest/docs/ops/configuration/traffic-management/traffic-routing/#unmatched-traffic

Without Istio, the ports field of a headless service is not strictly required because requests go directly to pod IPs, which can accept traffic on all ports. However, with Istio the port must be declared in the Service, or it will not be matched.

**Comment by guignome** [comment]
Similar issue: https://github.com/artemiscloud/activemq-artemis-operator/discussions/432

Also, I tried to manually add port 7800 to the headless service but the operator overwrites it.

**Reply by brusdev** [reply]
> Similar issue: #432
> 
> Also, I tried to manually add port 7800 to the headless service but the operator overwrites it.

This behavior is expected.

**Comment by brusdev** [comment, marked as answer]
@ramu11 I see your point, your proposal to add the port 7800 to the headless service makes sense to me:
https://istio.io/latest/docs/ops/configuration/traffic-management/traffic-routing/#headless-services
```
However, to support headless services, listeners are set up for each IP:Port pair in the headless service.
```

Feel free to create a new feature issue: https://github.com/artemiscloud/activemq-artemis-operator/issues

**Reply by ramu11** [reply]
@brusdev  created :https://github.com/artemiscloud/activemq-artemis-operator/issues/503

**Reply by brusdev** [reply]
@ramu11 thanks, contributions are always welcome 😀, are you planning to contribute to the issue #503 creating a pull request?
